### PR TITLE
Edge case platforms support (SmartOS, AIX, and gzip fallback)

### PR DESCRIPTION
**Pull request by refack**
Fixes: https://github.com/jasongin/nvs/issues/114
1. edge case `uname` parsing and workaround tar limitations
2. exclude AIX from `xz` detection, and add gzip download fallback

### CODE REVIEW

**Comment by jasongin** [comment]
> remove auto XZ upgrade

Discussion in the referenced issue doesn't appear to be settled, so it seems premature to drop all use of XZ packages. They are significantly smaller, so this would have a performance impact for all users.

Also there is a CI failure on Git Bash on Windows, where it is trying to download `https://nodejs.org/dist/v10.12.0/node-v10.12.0-win-.7z`

**Comment by refack** [comment]
> Discussion in the referenced issue doesn't appear to be settled, so it seems premature to drop all use of XZ packages. They are significantly smaller, so this would have a performance impact for all users.

Ack. I'll just exclude AIX.

> Also there is a CI failure on Git Bash on Windows, where it is trying to download `https://nodejs.org/dist/v10.12.0/node-v10.12.0-win-.7z`

Looking into that, I'm also waiting on access to AIX and SmartOS for more testing.

**Comment by refack** [comment]
* Restored `xz` promotion logic.
* Added a retry download
* Fixed `sed` replacing.

Both CIs are green, PTAL.

File: nvs.sh [hunk 1]
```diff
@@ -75,6 +80,18 @@ nvs() {
 			curl -L -# "${NODE_URI}" -o "${NODE_ARCHIVE}"
 		fi
 
+		if [ ! -f "${NODE_ARCHIVE}" ] && [ "${NODE_ARCHIVE_EXT}" = ".tar.xz" ]; then
+			NODE_ARCHIVE_EXT=".tar.xz"
```
Comment (jasongin) [inline nvs.sh]: I think this was intended to switch to `.tar.gz`?

File: nvs.sh [hunk 2]
```diff
@@ -75,6 +80,18 @@ nvs() {
...
+			TAR_FLAGS="-Jxvf"
```
Comment (jasongin) [inline nvs.sh]: I think it should be `-zxvf` for gzip instead of xzip extraction.

File: nvs.sh [hunk 3]
```diff
@@ -59,7 +59,12 @@ nvs() {
 		fi
 
 		# Download a node binary to use to bootstrap the NVS script.
-		local NODE_ARCH="$(uname -m | sed 's/x86_64/x64/' | sed 's/i686/x86/')"
+		# SmartOS (SunOS) reports `i86pc` which is synonymous with both x86 and x64.
+		local NODE_ARCH="$(uname -m | sed -e 's/x86_64/x64/;s/i86pc/x64/;s/i686/x86/')"
+		# On AIX reports `uname -m` reports the machine ID number of the hardware running the system.
+		if [ "${NVS_OS}" = "aix" ]; then
+			local NODE_ARCH="ppc64"
```
Comment (jasongin) [inline nvs.sh]: nit: `local` is not needed here since the local variable is already declared above.

**Comment by jasongin** [comment]
Closing because this is superseded by #169

**Pull request closed without merging**